include/zephyr/drivers/gnss/ublox_neo_m8_defines.h

```diff
@@ -0,0 +1,222 @@
+/** @file
+ * @brief NEO-M8 gnss module public API definitions header file.
+ *
+ * Copyright (c) 2022 Abel Sensors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_GNSS_UBLOX_NEO_M8_DEFINES_H_
+#define ZEPHYR_INCLUDE_DRIVERS_GNSS_UBLOX_NEO_M8_DEFINES_H_
+
+enum interface_description {
+	NBYTES_HIGH_ADDR = 0xFD,
+	NBYTES_LOW_ADDR = 0xFE,
+	DATA_STREAM_ADDR = 0xFF,
+
+	MAX_NMEA_SIZE = 83,
+	MAX_PAYLOAD_SIZE = 256,
+
+	NOACK = 150,
+	ACK = 151,
+
+	UBX_CLASS_NAV = 0x01,
```

Comment:
This used to be a file full of defines, however ycsin suggested making this file an enum instead of preprocessed macros.

Reply:
right but this is obviously wrong right?

Reply:
These values are for a specific protocol ublox has created called UBX. A ubx frame has a specific structure consisting of 2 header bytes, a class byte, an ID byte, 2 length bytes, a payload and 2 checksum bytes. The UBX_CLASS_NAV is a class with number 0x01. This message would regard the navigational settings. The UBX_ACK_ACK is the value of the payload of an acknowledgement frame. Used to check if the message has been successfully sent and processed. I have included this in the file because it makes it more clear what is happening within the code.

Reply:
Are those enums (with the same value) referring to two different things? Then they must go in two different enums.

Reply:
Ah okay, I'll change that then